### PR TITLE
add num active cells

### PR DESCRIPTION
**Pull request by fgfuchs**
add total  number of active cells to log

### CODE REVIEW

**Comment by akva2** [comment]
you can trigger builds here now.

**Comment by atgeirr** [comment]
Sorry for merging a bit quickly the other one...

For this, I have the usual formatting nitpick (space around binary operators). I agree we should automate, progress on that has stalled for a while, but there is a plan.

File: opm/grid/cpgrid/CpGrid.cpp [hunk 1]
```diff
@@ -252,7 +252,9 @@ CpGrid::scatterGrid(EdgeWeightMethod method, const std::vector<cpgrid::OpmWellTy
         cc.sum(cc_num_cells.data(),cc.size());
 
         if (cc.rank() == 0) {
-            std::string str = "\nAfter loadbalancing on " + std::to_string(cc.size()) + " processes we have\nrank: num cells\n----------------\n";
+            std::string str = "\nLoad balancing distributes "+std::to_string(data_->size(0))+" active cells on " + std::to_string(cc.size()) + " processes as follows:\n";
+            str += "rank: num cells\n";
```
Comment (blattms) [inline opm/grid/cpgrid/CpGrid.cpp]: active missing?

Comment (fgfuchs) [reply opm/grid/cpgrid/CpGrid.cpp]: Could you please explain what you mean?

**Comment by fgfuchs** [comment]
jenkins build this please

**Comment by blattms** [comment]
" rank: num cells" -> "rank: num active cells"

On second thought, I must admit that the active word here and above does not make any sense as there are no inactive cells in CpGrid.

**Comment by alfbr** [comment]
> the active word here and above does not make any sense as there are no inactive cells in CpGrid.

True, but that is an implementation detail, which users are probably unaware of. Since this is a user facing message, I would actually suggest making it more explicit, something like:
"number of active cells in partition"

**Comment by atgeirr** [comment]
This is what I want:
```
rank   owned cells   overlap cells   total cells
------------------------------------------------
   0          4000            1200          5200
   1          4000             400          4400
   2         10000            2000         12000
   3          7000             700          7700
------------------------------------------------
 sum         25000            4300         29300
```
The sum of owned cells should equal the total number of active cells (printed above). Keep both, it's educational...

**Comment by fgfuchs** [comment]
jenkins build this please

**Comment by fgfuchs** [comment]
Example output for norne
Load balancing distributes 44431 active cells on 6 processes as follows:
  rank   owned cells   overlap cells   total cells
--------------------------------------------------
     0          8086            1348          9434
     1          6852            3461         10313
     2          6714             251          6965
     3          8143            2648         10791
     4          7816            2052          9868
     5          6820            1089          7909
--------------------------------------------------
   sum         44431           10849         55280

File: opm/grid/cpgrid/CpGrid.cpp [hunk 2]
```diff
@@ -245,21 +245,94 @@ CpGrid::scatterGrid(EdgeWeightMethod method, const std::vector<cpgrid::OpmWellTy
...
+                    str_tmp.insert(str_tmp.begin(), 16 - str_tmp.length(), ' ');
+                    str += str_tmp;
+                }
+                // total cells
+                {
+                    auto str_tmp = std::to_string(cc_total_cells[i]);
+                    str_tmp.insert(str_tmp.begin(), 14 - str_tmp.length(), ' ');
+                    str += str_tmp;
+                }
+                str += "\n";
```
Comment (bska) [inline opm/grid/cpgrid/CpGrid.cpp]: I'm sure this is correct, and the output certainly looks good, but I worry about the maintainability of these constructs.  Did you consider using `ostringstream` and `iomanip` here?  For instance, something along the lines of
```C++
std::ostringstream line;

line << std::setw(6) << i
     << std::setw(14) << cc_owned_cells[i]
     << std::setw(16) << cc_overlap_cells[i]
     << std::setw(14) << cc_total_cells[i] << '\n';

str += line.str();
```
possibly with some additional constants to define the field widths?

Comment (fgfuchs) [reply opm/grid/cpgrid/CpGrid.cpp]: Looks better. I will change it.

**Comment by fgfuchs** [comment]
jenkins build this please

**Comment by atgeirr** [comment]
Looks good, merging.

**Pull request merged**